Velocity-Implicit Euler Fixed Step - Second PR of #12528

antequ opened the pull request:
This PR adds a fixed-step implementation of the new first-order Velocity-Implicit Euler Integrator (Issue #12528). This is the second PR out of four.

The documentation is based on the equations listed here: https://docs.google.com/document/d/118J6rkR9ghh2_3WY0O5EDXhfkgMOXXqTQ2usrhICv30/edit


The current state of this PR also contains all the changes of #12532 together with it; this PR is a branch on top of the other PR. Do not review the files and code from that PR; I've marked them for your convenience. (The simplest way to filter is just look at the changes from the latest commit, 8ef182c. ) After that PR is merged, this PR should contain less than 1000 lines of changes. For the purposes of reviewing, the contents of this PR do not depend on understanding the other PR.

+@edrumwri +@sherm1 +@amcastro-tri

I've marked this "Do not merge", but feel free to start reviewing!

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/12543)
<!-- Reviewable:end -->

jwnimmer-tri commented:
> It seems to be passing the CI in reviewable, so I assume those pass.

Pre-merge CI does not run all variations.  You have to opt-in.
See https://drake.mit.edu/jenkins.html.

@drake-jenkins-bot linux-bionic-clang-bazel-experimental-everything-valgrind-memcheck please
@drake-jenkins-bot linux-bionic-clang-bazel-experimental-valgrind-memcheck please
@drake-jenkins-bot linux-bionic-gcc-bazel-experimental-everything-valgrind-memcheck please
@drake-jenkins-bot linux-bionic-gcc-bazel-experimental-valgrind-memcheck please

antequ commented:
@drake-jenkins-bot linux-bionic-clang-bazel-experimental-everything-valgrind-memcheck please @drake-jenkins-bot linux-bionic-clang-bazel-experimental-valgrind-memcheck please @drake-jenkins-bot linux-bionic-gcc-bazel-experimental-everything-valgrind-memcheck please @drake-jenkins-bot linux-bionic-gcc-bazel-experimental-valgrind-memcheck please





<!-- Sent from Reviewable.io -->

antequ commented:
*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/12543#-M-0_lao9IRt8WFTuhR8:-M-Vrhlu3b820mPdT9ee:b-s2hsvx) (no related file):*
<details><summary><i>Previously, antequ (Ante Qu) wrote…</i></summary><blockquote>

Addressed most recent review comments. @drake-jenkins-bot linux-bionic-clang-bazel-experimental-everything-valgrind-memcheck please @drake-jenkins-bot linux-bionic-clang-bazel-experimental-valgrind-memcheck please @drake-jenkins-bot linux-bionic-gcc-bazel-experimental-everything-valgrind-memcheck please @drake-jenkins-bot linux-bionic-gcc-bazel-experimental-valgrind-memcheck please

Adding +@sherm1 for platform review. In the meantime I'll squash and rebase.
</blockquote></details>

Squashed and rebased!

 @drake-jenkins-bot linux-bionic-clang-bazel-experimental-everything-valgrind-memcheck please @drake-jenkins-bot linux-bionic-clang-bazel-experimental-valgrind-memcheck please @drake-jenkins-bot linux-bionic-gcc-bazel-experimental-everything-valgrind-memcheck please @drake-jenkins-bot linux-bionic-gcc-bazel-experimental-valgrind-memcheck please

---


<!-- Sent from Reviewable.io -->
